phoenix-tools/src/main/java/org/apache/phoenix/schema/SchemaExtractionProcessor.java

```diff
@@ -203,6 +203,9 @@ protected String extractCreateViewDDL(PTable table) throws SQLException {
     private String generateCreateViewDDL(String columnInfoString, String baseTableFullName,
             String whereClause, String pSchemaName, String pTableName) {
         String viewFullName = SchemaUtil.getQualifiedTableName(pSchemaName, pTableName);
+        if(Character.isDigit(pTableName.charAt(0))) {
```

Comment:
Is this a bug in SchemaUtil.getQualfiedTableName? This seems like something we shouldn't have to repeat everywhere.

Reply:
I think not, this is mainly for parsing. For Internals of phoenix, function should ignore quotes around the name.

Reply:
If you look at the phoenix grammar for create table or create view the table name is of the format below: https://phoenix.apache.org/language/index.html#table_ref
https://phoenix.apache.org/language/index.html#name

So table names can begin with [A-Z] or _ so any table name that doesn't begin with this has to begin with a quote. So just checking for `isDigit` is wrong as we might miss other cases where the first letter is something other than a digit for example like a `$` or other character. The fix should be if the first character is not a letter or  underscore then quote.

Reply:
Makes sense! Let me correct it.